x-pack/plugins/apm/common/chart_preview.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { isLeft } from 'fp-ts/lib/Either';
+import * as t from 'io-ts';
+import moment from 'moment';
+
+const getToEnumRt = <T>(enumObj: T, enumName = 'enum') =>
+  new t.Type<T[keyof T], string>(
+    enumName,
+    (u): u is T[keyof T] => Object.values(enumObj).includes(u),
+    (u, c) =>
+      Object.values(enumObj).includes(u)
+        ? t.success(u as T[keyof T])
+        : t.failure(u, c),
+    (a) => (a as unknown) as string
+  );
+
+export enum TIME_UNITS {
+  SECOND = 's',
+  MINUTE = 'm',
+  HOUR = 'h',
+  DAY = 'd',
+}
+
+const toTimeUnitRt = getToEnumRt(TIME_UNITS, 'timeUnitRt');
+
+const BUCKET_SIZE = 20;
+
+export function getIntervalAndTimeRange({
```

Comment:
Hmm... Why is this in common? Isn't it only used on the client?

Reply:
... Either way, do we really need to validate it? I thought this was happening on the server but if it's on the client, then we receive the `windowUnit` straight from another app. As long as we trust that app e should be fine.